### ISSUE TITLE
Mod doesn't work (MissingFieldException)

### ISSUE THREAD

**Issue by KingEnderBrine** [cheeeeeeeeeen/RoR2-ChensMinionRetarget#4]
### Describe the Bug
1. They have changed `currentEnemy` from field to property. So the mod doesn't work at all.
```
[Error  : Unity Log] MissingFieldException: Field 'RoR2.CharacterAI.BaseAI.currentEnemy' not found.
Stack trace:
DMD<>?1167836672.Hook<RoR2.CharacterAI.BaseAI::EvaluateSkillDrivers>?-501121024 (RoR2.CharacterAI.BaseAI ) (at <873019a913634471bdd11a8386c8ca1a>:IL_0014)
DMD<>?1167836672.Glue:ThiscallStructRetPtr<RoR2.CharacterAI.BaseAI::EvaluateSkillDrivers,DMD<>?1167836672::Hook<RoR2.CharacterAI.BaseAI::EvaluateSkillDrivers>?-501121024> (RoR2.CharacterAI.BaseAI , RoR2.CharacterAI.BaseAI+SkillDriverEvaluation& ) (at <ac0af3cef06f4a9492f338fe1c870550>:IL_0000)
RoR2.CharacterAI.BaseAI.FixedUpdate () (at <da7c19fa62814b28bdb8f3a9223868e1>:IL_014F)
```
2. Also I'm currently working on my mod that has the same mechanics as this mod but as a skill. Was wondering is this line correct?
https://github.com/cheeeeeeeeeen/RoR2-ChensMinionRetarget/blob/a77ef980d7275249a62edf82e6fc396b3aa11b08/ChensMinionRetarget/MinionRetarget.cs#L66
Seems like it should be:
```cs
if (skillDriverEval.dominantSkillDriver.maxDistanceSqr > (targetPosition - selfPosition).sqrMagnitude && self.HasLOS(targetPosition)) 
```
At least from my testing, I couldn't get it to work properly otherwise.

### Console log
[LogOutput.log](https://github.com/cheeeeeeeeeen/RoR2-ChensMinionRetarget/files/6378031/LogOutput.log)

**Comment by cheeeeeeeeeen** [comment]
Hello KEB, Thanks for the heads up, I'm not sure when I can get back on this, but it does look like an easy fix.

For the 2nd item, Your proposed solution has a different behavior of course. I only want them to retarget if they are in range of the minion; your solution would make them retarget even if they're outside the max range. From what I remember, this makes the AI choose again if they're outside the range, but thus interrupting their current order of attack.

**Comment by KingEnderBrine** [comment]
Maybe I'm missing something, but `skillDriverEval.dominantSkillDriver.maxDistanceSqr` is the range of minion and `(targetPosition - selfPosition).sqrMagnitude` is distance between minion and target. So if you want to target only in a range you need the range to be bigger than the distance. So, your solution is the opposite of what you want.

**Comment by cheeeeeeeeeen** [comment]
I get it now. Yeah, the opposite of what I have now (your proposed solution) is *actually what I want.*

**Comment by cheeeeeeeeeen** [comment]
Fixed by abe4a205f0c73cdf08c68927955c11446c284e7e. Bad commit message, my bad.
